pkg/helm/helm_helpers.go

```diff
@@ -71,7 +79,21 @@ func ComponentsFromManifest(manifest string) []Component {
 			log.Error().Print("failed to Unmarshal k8s obj", field.M{"Error": err})
 			continue
 		}
-		ret = append(ret, Component{k8sType: K8sObjectType(strings.ToLower(tmpK8s.ObjKind)), name: tmpK8s.MetaData.Name})
+		k8sType := K8sObjectType(strings.ToLower(tmpK8s.ObjKind))
+		resourceName := tmpK8s.MetaData.Name
+		// Either append all components or filter using the filter func
+		if filter == nil || filter(resourceName, k8sType) {
+			ret = append(ret, Component{k8sType: k8sType, name: resourceName, originalDump: objYaml})
+		}
 	}
 	return ret
 }
+
+func GetObjectFromComponent[T runtime.Object](component Component) (T, error) {
```

Comment:
```suggestion
func GetK8sObjectsFromRenderedResource[T runtime.Object](component Component) (T, error) {
```
`GetK8sObjectsFromRenderedManifest`

Reply:
Done